### PR TITLE
Update gdb docs for new DWARF enhancements.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title. The new DWARF enhancements for 0.55 warrant an update to the docs for gdb.

### CODE REVIEW

File: docs/source/user/troubleshoot.rst [hunk 1]
```diff
@@ -604,20 +613,29 @@ presented for brevity):
...
 function (this is the Numba internal symbol registered as breakpoint with
-``gdb``). Issuing a ``step`` at this point moves to the stack frame of the
+``gdb``). Issuing a ``step`` twice at this point moves to the stack frame of the
 compiled Python source. From there, it can be seen that the variables ``a`` and
 ``b`` have been evaluated but ``c`` has not, as demonstrated by printing their
 values, this is precisely as expected given the location of the ``gdb()`` call.
-Issuing a ``next`` then evaluates line ``7`` and ``c`` is assigned a value as
-demonstrated by the final print.
+Issuing a ``break`` on line 9 and then continuing evaluates line ``7`` and ``c``
+is assigned a value as demonstrated by ``info locals`` when the breakpoint it
```
Comment (sklam) [inline docs/source/user/troubleshoot.rst:637]: I have a hard time parsing the "continuing evaluates ... is assigned a value"

**Review by sklam — COMMENTED** [review]
Thanks for the patch!

I have one comment:

File: docs/source/user/troubleshoot.rst (hunk 1 above)
Comment (stuartarchibald) [reply docs/source/user/troubleshoot.rst]: I've reworded this in cf6dee6

Comment (stuartarchibald) [reply docs/source/user/troubleshoot.rst]: And again, as discussed OOB in 6a9c8a1 .

**Review by sklam — APPROVED** [review]
Perfect!

**Comment by esc** [comment]
@stuartarchibald thanks for the patch! @sklam thank you for the review!

**Pull request merged**